controller/src/main/java/io/pravega/controller/store/stream/OperationNotAllowed.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+import io.pravega.controller.retryable.RetryableException;
+
+/**
+ * Exception thrown attempting an operation that is not allowed to be performed at the moment.
+ */
+public class OperationNotAllowed extends RuntimeException implements RetryableException {
```

Comment:
We need some retryablity vs non retryability and some other traits so merely have them as type in StoreException may not suffice.
We should look at how we can update add traits to store exception so that predicates can be defined in a generic manner against them. 
lets take that as a separate issue. (opened issue #1367)